test/e2e/e2e.go

```diff
@@ -719,19 +650,44 @@ var _ = ginkgo.Describe("e2e control plane", func() {
 	f := wrappedTestFramework(svcname)
 
 	var (
-		extDNSIP              string
-		numControlPlanePods   int
-		controlPlanePodName   string
-		controlPlaneLeaseName string
-		nodes                 []v1.Node
+		extDNSIP                   string
+		numControlPlanePods        int
+		controlPlanePodName        string
+		controlPlaneLeaseName      string
+		providerCtx                infraapi.Context
+		secondaryProviderNetwork   infraapi.Network
+		secondaryExternalContainer infraapi.ExternalContainer
 	)
 
 	ginkgo.BeforeEach(func() {
+		var err error
+		providerCtx = infraprovider.Get().NewTestContext()
+		cmdArgs := []string{"network", "create", "--driver", "bridge", "--subnet", secondaryIPV4Subnet, secondaryNetworkName}
```

Comment:
this seems provider specific doesn't it? Shouldn't we be abstracting this as well?